fabric-dimensions-v1/src/main/java/net/fabricmc/fabric/mixin/dimension/MixinWorld.java

```diff
@@ -0,0 +1,20 @@
+package net.fabricmc.fabric.mixin.dimension;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.Dimension;
+import net.minecraft.world.dimension.DimensionType;
+
+@Mixin(World.class)
+public abstract class MixinWorld {
+	// This redirector un-hardcodes the overworld checks in World.isDay() and World.isNight()
+	@Redirect(method = {"isDay", "isNight"}, at = @At(value = "INVOKE", target = "Lnet/minecraft/world/dimension/Dimension;getType()Lnet/minecraft/world/dimension/DimensionType;"))
+	private DimensionType replaceCustomDimensionsWithOverworld(Dimension dimension) {
+		// Replaces the dimension type with overworld if the dimension has a visible sky.
+		// hasVisibleSky() is also used for time checking in other places like clocks, for example.
```

Comment:
The comment doesn't explain the rationale, it should say something like `The calling code enables the day/night cycle for DimensionType.OVERWORLD, pretend to be that if hasVisibleSky() is true. hasVisibleSky() is true for overworld, false for nether and end, and can be customized by the dimension. It is already used for time checking in other places like clocks, for example.`

Reply:
I improved the comment a bit in 264ce5d.